ncpi_model_forge/ingest_plugin/ncpi_patient.py

```diff
@@ -0,0 +1,168 @@
+                    "system": "http://fhir.kids-first.io/CodeSystem/species",
+                    "code": "448771007",
+                    "display": "Canis lupus subspecies familiaris",
+                }
+            ],
+            "text": "Dog",
+        },
+    },
+    "Human": {
+        "url": "http://fhir.kids-first.io/StructureDefinition/species",
+        "valueCodeableConcept": {
+            "coding": [
+                {
+                    "system": "http://fhir.kids-first.io/CodeSystem/species",
+                    "code": "337915000",
+                    "display": "Homo sapiens",
+                }
+            ],
+            "text": "Human",
+        },
+    },
+}
+
+
+class Patient:
+    class_name = "patient"
+    resource_type = "Patient"
+    target_id_concept = "PATIENT_FHIR_ID"
+
+    @staticmethod
+    def build_key(record):
+        assert None is not record["PATIENT_ID"]
+        return record["PATIENT_ID"]
+
+    @staticmethod
+    def build_entity(record, key, get_target_id_from_record):
+        study_id = record.get("STUDY_ID")
+        participant_id = record.get("PATIENT_ID")
+        ethnicity = record.get("PATIENT_ETHNICITY")
+        race = record.get("PATIENT_RACE")
+        species = record.get("PATIENT_SPECIES")
+        gender = record.get("PATIENT_GENDER")
+
+        entity = {
+            "resourceType": Patient.resource_type,
+            # This is how you would use get_target_id_from_record, but we're going
+            # to skip it here.
+            # "id": get_target_id_from_record(Patient, record),
+            "meta": {
+                "profile": [
+                    "http://fhir.ncpi-project-forge.io/StructureDefinition/ncpi-patient"
+                ]
+            },
+            "identifier": [
+                {
+                    "system": "urn:ncpi-project-forge:unique-string",
+                    "value": f"Patient__{study_id}__{key}",
+                },
+            ],
+        }
+
+        if ethnicity:
+            if omb_ethnicity_category.get(ethnicity):
+                entity.setdefault("extension", []).append(
+                    {
+                        "url": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-ethnicity",
+                        "extension": [
+                            omb_ethnicity_category[ethnicity],
+                            {"url": "text", "valueString": ethnicity},
+                        ],
+                    }
+                )
+
+        if race:
+            if omb_race_category.get(race):
+                entity.setdefault("extension", []).append(
+                    {
+                        "url": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-race",
+                        "extension": [
+                            omb_race_category[race],
+                            {"url": "text", "valueString": race},
+                        ],
+                    }
+                )
+
+        if species:
```

Comment:
I don't know if we should include any of the species stuff yet because the species extension isn't merged into master yet

Reply:
It was described in the guide though?